### PR TITLE
Banner custom name API

### PR DESCRIPTION
**Pull request by Swedz**

### CODE REVIEW

File: Spigot-API-Patches/0317-Banner-custom-name-API.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+From: Swedz <me@swedz.net>
+Date: Fri, 18 Jun 2021 01:20:44 -0400
+Subject: [PATCH] Banner custom name API
+
+
+diff --git a/src/main/java/org/bukkit/block/Banner.java b/src/main/java/org/bukkit/block/Banner.java
+index e6eb3c04b9fa6cd232f95dd5851d365234972efd..5f9a3710825cc84ab1276d7e26707ebdfcbdd308 100644
+--- a/src/main/java/org/bukkit/block/Banner.java
++++ b/src/main/java/org/bukkit/block/Banner.java
+@@ -83,4 +83,25 @@ public interface Banner extends TileState {
```
Comment (Machine-Maker) [inline Spigot-API-Patches/0317-Banner-custom-name-API.patch]: This needs to extend the bukkit Nameable interface.

**Review by Machine-Maker — CHANGES_REQUESTED** [review]
This should also target the 1.17 branch, as probably no new 1.16.5 API will be added due the massive difference in the build system and mappings making it difficult to port over patches.

File: Spigot-API-Patches/0317-Banner-custom-name-API.patch (hunk 1 above)
Comment (Swedz) [reply Spigot-API-Patches/0317-Banner-custom-name-API.patch]: Just did the Nameable interface bit. Not sure how to go about swapping this to 1.17

Comment (Swedz) [reply Spigot-API-Patches/0317-Banner-custom-name-API.patch]: I also just realized I screwed up the commit. I had copied some code from the Beacon implementation for Nameable and didn't remove the unused imports. Fixing that now

**Review by Machine-Maker — CHANGES_REQUESTED** [review]
Needs to be rebased for 1.17.1

**Comment by Owen1212055** [comment]
Are you interested in continuing this PR? I am willing to continue progress on this.

**Comment by Swedz** [comment]
> Are you interested in continuing this PR? I am willing to continue progress on this.

No I am not, I apologize for the late response. Feel free to take this up.

**Comment by Machine-Maker** [comment]
Superseded by https://github.com/PaperMC/Paper/pull/7711

**Pull request closed without merging**